### ISSUE TITLE
Pronouns as/with determiners: "we/you/%them pilots", "you all", "you two"

### ISSUE THREAD

**Issue by nschneid** [UniversalDependencies/UD_English-EWT#436]
I can't find an issue for this, though it came up in the [Mischievous Nominals paper](https://aclanthology.org/2021.udw-1.14) - example (5):

- **We** pilots deserve a pay raise
- **You** guys deserve a pay raise

Standardly, it is just "we/us" and "you" that do this, though dialectal variants are possible like "keep **them** lovely kebabs coming" (EWT).

Arguably, "we/you" are acting as determiners, making the nominal definite and adding person features. (This is the CGEL analysis.) EWT follows this analysis: [EWT PRON-as-`det`](https://universal.grew.fr/?custom=65177e4f2bcfc)

But this is rare, and annotators/users might be surprised to see (what look like) pronouns in `det` position.

GUM has `dep(guys, you)`.

The MNCs paper suggests `nmod:desc`.

Some other related things:

- you idiot, you absolute fool, etc. (singular noun but I don't think these can be subjects) ([Twitter discussion](https://twitter.com/rm_gibson/status/1707331179953660074))
- you all/both - currently `det(you, all)`. [EWT PRON-headed `det`](https://universal.grew.fr/?custom=65177e0f9e4e5)
- us three (1 EWT token), you three (1 GUM token) - currently `nummod`

**Comment by amir-zeldes** [comment]
dep in GUM is intentional and generally applies as a placeholder for things we wanted to be nmod:desc (until whenever we decide to implement that, if we do)

**Comment by nschneid** [comment]
In the end we defined nmod:desc to be specific to name-modifiers. Let's go with nmod:unmarked(we, pilots), nmod:unmarked(you, all).

**Comment by nschneid** [comment]
OK now I'm wondering how to disambiguate "all"/"both" as attaching to the pronoun vs. the following predicate. This is currently det(We, all) in EWT:

- We all know what happened.

But I don't see how this is different from

- The students all know what happened.

Perhaps with "all" and "both", we should favor predicate attachment if the pronoun is a subject, and only attach to the pronoun if it is an object or oblique?

- I look forward to seeing **you all** there.
- You'll bring honor to **us all**.

**Comment by nschneid** [comment]
Implemented above:

```grew
rule r0 { % you guys
    pattern {
      X[lemma=we|us|you]; 
      e: Y-[det]->X;
      e2: Y-[E:det]->X;
    }
    commands { 
      del_edge e;
      del_edge e2;
      add_edge X-[nmod:unmarked]->Y;
      add_edge X-[E:nmod:unmarked]->Y;
      shift_in Y ==> X;
      shift_out Y ==> X;
    }
}
rule r1 { % us two
    pattern {
      X[lemma=we|us|you]; 
      e: X-[nummod]->Y; 
      e2: X-[E:nummod]->Y;
      X << Y;
    }
    commands { 
      del_edge e;
      del_edge e2;
      add_edge X-[nmod:unmarked]->Y;
      add_edge X-[E:nmod:unmarked]->Y;
    }
}
rule r2 { % you all (non-subj)
    pattern {
      X[upos=PRON]; 
      e: X-[det]->Y; 
      e2: X-[E:det]->Y;
      X << Y;
    } without { Z-[nsubj]->X }
    commands { 
      del_edge e;
      del_edge e2;
      add_edge X-[nmod:unmarked]->Y;
      add_edge X-[E:nmod:unmarked]->Y;
 }
}
rule r3 { % you all (subj)
    pattern {
      X[upos=PRON]; 
      e: X-[det]->Y; 
      e2: X-[E:det]->Y;
      X << Y;
      Z-[nsubj]->X;
    }
    commands {
      del_edge e;
      del_edge e2;
      add_edge Z-[advmod]->Y;
      add_edge Z-[E:advmod]->Y;
 }
}
rule r5 { % keep them lovely kebabs coming: leave det(kebabs, them) but mark "them" as Style=Vrnc
    pattern {
      X[lemma=they]; 
      Y-[det]->X;
    }
    commands { 
      X.Style="Vrnc";
    }
}
```

**Comment by nschneid** [comment]
"Them" as in "keep them kebabs coming" is now DET/DT/Style=Vrnc with lemma "them" ("they" cannot be used in this way).

**Comment by AngledLuffa** [comment]
Not slang for `those`?

**Comment by nschneid** [comment]
I put vernacular instead of slang but am not really sure how to draw the line. Yes, the standard form is "those".

**Comment by AngledLuffa** [comment]
Agreed, and that makes me think `that` would be the right lemma, not `them`

**Comment by nschneid** [comment]
If the lemma is where we'd expect it to be listed in the dictionary, I wouldn't expect to find it under "that". Merriam-Webster has it [under "them"](https://www.merriam-webster.com/dictionary/them).

**Comment by AngledLuffa** [comment]
Ok, that's fair.  I was thinking of it as actually being the word `that`, just spoken in a vernacular sense, whereas this dictionary is treating it as an actual usage of the word `them`.  Works for me

**Comment by AngledLuffa** [comment]
This change would apply to `we all` in a context such as these in ParTUT?

```
I still have a vivid memory, as, I believe, we all do, of the catastrophe
We all regret that the European Committee for Standardisation
We all agree that we should ask that ...
... but things that we all pretty well know
about the efforts we Ukrainians are now making to defend our country and our democracy
```

dev:

```
We all know the Earth is in trouble
```

**Comment by nschneid** [comment]
One approach is to say that if "we" can be replaced with "the students", "all" should be advmod of the predicate. I think that applies to all the ParTUT "we all" sentences.

For "we Ukrainians": nmod:unmarked(we, Ukrainians)

**Comment by AngledLuffa** [comment]
Excellent, thanks!

**Comment by nschneid** [comment]
#598 identified cases where `appos` was used for "you guys" and similar, instead of `det`. So we need an additional rule to convert them to `nmod:unmarked`:

```
rule r0b { % you guys
    pattern {
      X[lemma=we|us|you]; 
      e: X-[appos]->Y;
      e2: X-[E:appos]->Y;
    }
    commands { 
      del_edge e;
      del_edge e2;
      add_edge X-[nmod:unmarked]->Y;
      add_edge X-[E:nmod:unmarked]->Y;
      shift_in Y ==> X;
      shift_out Y ==> X;
    }
}
```
